### ISSUE TITLE
3.7: instance member function becomes instance member property in type references [outcome: fixed since 3.7.0-dev.20191016]

### ISSUE THREAD

**Issue by bre1470** [microsoft/TypeScript#33747]
**TypeScript Version:**  3.7.0-dev.20191002

**Search Terms:** instance member property

**Code**

```ts
declare global {
    class A {
        public f(): void;
    }
    class BA extends A {
        public f(): void;
    }
    class CA extends A {
        public f(): void;
    }
    class D extends CA {
        public f: BA['f']; // errors with 3.7
    }
}

export default undefined;
```

**Expected behavior:** use declarations without error

**Actual behavior:** errors with `Class 'CA' defines instance member function 'f', but extended class 'D' defines it as instance member property.`

**Playground Link:** https://www.typescriptlang.org/play/?ts=Nightly#code/CYUwxgNghgTiAEBzCB7ARlC8DeAoeB8kUAzifAII76G0AOArmhAJZjwBmAFAJQBc8AG4oWwANw0CAX0lFoZeACEqIAB4AXEADtg5Knlr0mrdt35CR42TNrEFAYRUbtuytUOFGzNp14DhohK0NoR25AAi8GqaOuSO7h7wXiacAsoA2gDkHJkAumLwAPSFUTAwKDDkAO4s6gAW8ADMAHQA7Na4MrhgKFokKBAgzaiIXJkAEiAQqAA08FUVEMAAhJk8ErhqdBXq8KAcUAwQuww6IBwsWiBWQA (compiles there without error)

**Related Issues:** -

**Comment by NoPhaseNoKill** [comment]
I'm getting the same issue, namely when using react lifecycle methods that have an arrow function.

componentWillMount = () => {...}

/*Throws 'Class component'' Props<State, any>  defines instance member function componentDidMount, but extended class 'X ' defines it as an instance member property
*/

Error is fixed with:
componentWillMount() {....}

The problem with the workaround, is that to my knowledge it will change what is bound to context of 'this' under certain scenarios. Could someone please clarify on this?

**Comment by bre1470** [comment]
In your case the `this` scope might change as arrow functions use the outer scope as they do not have their own.

**Comment by seansfkelley** [comment]
This is still the case in `3.7.0-beta`. Original issue: https://github.com/Microsoft/TypeScript/issues/17113.

**Comment by seansfkelley** [comment]
Looks like this is fixed in nightly already, but I haven't tested it myself: https://github.com/microsoft/TypeScript/pull/34494

**Comment by bre1470** [comment]
Confirmed. The issue is solved for me as well since `3.7.0-dev.20191016`.
